diff --git a/src/utils/polyfill.rs b/src/utils/polyfill.rs
--- a/src/utils/polyfill.rs
+++ b/src/utils/polyfill.rs
@@ -371,6 +371,8 @@ pub fn wpreftype_to_type(ty: wasmparser::RefType) -> WasmResult<Type> {
         Ok(Type::ExternRef)
     } else if ty.is_func_ref() {
         Ok(Type::FuncRef)
+    } else if ty == wasmparser::RefType::EXNREF || ty == wasmparser::RefType::NULLEXNREF {
+        Ok(Type::ExceptionRef)
     } else {
         Err(format!("Unsupported ref type: {:?}", ty))
     }
